### PR TITLE
Authentication - Use DB Users

### LINKED ISSUE #5: Add Authentication user details service
Need to add a user details service that fetches data from the DB.
Currently, this data is being fetched from a pre-configured in-memory store.

### PR DESCRIPTION
**Pull request by chrisvrose**
- Remove the in-memory users table, and fetch username details from the DB.
- Add unit tests and configuration for repositories.
- Disallow lombok from being part of coverage (by adding `lombok.config`)

Closes #5

### CODE REVIEW

**Review by Prajwalprakash3722 — COMMENTED** [review]
LGTM 👌🏼

**Review by Prajwalprakash3722 — APPROVED** [review]

**Comment by Prajwalprakash3722** [comment]
would be better to update README file

**Comment by chrisvrose** [comment]
Good point. Let me do that before this PR is merged

File: lombok.config [hunk 1]
```diff
@@ -0,0 +1,2 @@
+config.stopBubbling = true
```
Comment (chrisvrose) [inline lombok.config:1]: This file prevents lombok annotations from being considered in coverage. It doesn't make sense to perform UTs for things we haven't written.

**Comment by chrisvrose** [comment]
Expanded the scope of the PR to allow:
- Added repo tests. This will test some of the scenarios we have ~~and also indirectly add coverage of the entities~~.
- Added config for repository tests. An in-memory H2 db is used to run theses tests to ensure our written functionality is working well.

**Review by Prajwalprakash3722 — APPROVED** [review]

**Pull request merged**